src/league_gui.cpp

```diff
@@ -0,0 +1,451 @@
+			if (!Town::IsValidID(link.target)) return;
+			xy = Town::Get(link.target)->xy;
+			break;
+
+		case LT_COMPANY:
+			ShowCompany((CompanyID)link.target);
+			return;
+
+		case LT_STORY_PAGE: {
+			if (!StoryPage::IsValidID(link.target)) return;
+			CompanyID story_company = StoryPage::Get(link.target)->company;
+			ShowStoryBook(story_company, link.target);
+			return;
+		}
+
+		default: NOT_REACHED();
+	}
+
+	if (_ctrl_pressed) {
+		ShowExtraViewportWindow(xy);
+	} else {
+		ScrollMainWindowToTile(xy);
+	}
+}
+
+
+class ScriptLeagueWindow : public Window {
+private:
+	LeagueTableID table;
+	std::vector<std::pair<uint, const LeagueTableElement *>> rows;
+	uint rank_width;     ///< The width of the rank ordinal
+	uint text_width;     ///< The width of the actual text
+	uint score_width;    ///< The width of the score text
+	int line_height;     ///< Height of the text lines
+	Dimension icon_size; ///< Dimenion of the company icon.
+	std::string title;
+
+	/**
+	 * Rebuild the company league list
+	 */
+	void BuildTable()
+	{
+		this->rows.clear();
+		this->title = std::string{};
+		auto lt = LeagueTable::GetIfValid(this->table);
+		if (lt == nullptr) return;
+
+		/* We store title in the window class so we can safely reference the string later */
+		this->title = lt->title;
+
+		std::vector<const LeagueTableElement *> elements;
+		for(LeagueTableElement *lte : LeagueTableElement::Iterate()) {
+			if (lte->table == this->table) {
+				elements.push_back(lte);
+			}
+		}
+		std::sort(elements.begin(), elements.end(), [](auto a, auto b) { return a->rating > b->rating; });
+
+		uint rank = 0;
+		for (uint i = 0; i != elements.size(); i++) {
+			auto *lte = elements[i];
+			if (i > 0 && elements[i - 1]->rating != lte->rating) rank = i;
+			this->rows.emplace_back(std::make_pair(rank, lte));
+		}
+	}
+
+public:
+	ScriptLeagueWindow(WindowDesc *desc, LeagueTableID table) : Window(desc)
+	{
+		this->table = table;
+		this->BuildTable();
+		this->InitNested(table);
+	}
+
+	void SetStringParameters(int widget) const override
+	{
+		if (widget != WID_SLT_CAPTION) return;
+		SetDParamStr(0, this->title);
+	}
+
+	void OnPaint() override
+	{
+		this->DrawWidgets();
+	}
+
+	void DrawWidget(const Rect &r, int widget) const override
+	{
+		if (widget != WID_SLT_BACKGROUND) return;
+
+		auto lt = LeagueTable::GetIfValid(this->table);
+		if (lt == nullptr) return;
+
+		Rect ir = r.Shrink(WidgetDimensions::scaled.framerect);
+
+		if (!lt->header.empty()) {
+			SetDParamStr(0, lt->header);
+			ir.top = DrawStringMultiLine(ir.left, ir.right, ir.top, UINT16_MAX, STR_JUST_RAW_STRING, TC_BLACK);
+		}
+
+		int icon_y_offset = (this->line_height - this->icon_size.height) / 2;
+		int text_y_offset = (this->line_height - FONT_HEIGHT_NORMAL) / 2;
+
+		uint rank_left, rank_right, icon_left, text_left, text_right, score_left, score_right;
+		int spacer = WidgetDimensions::scaled.framerect.left + WidgetDimensions::scaled.framerect.right;
+		if (_current_text_dir == TD_RTL) {
```

Comment:
Without changing too much how it's all calculated, something like this should work (and handle both ltr & rtl):

```
Rect rank = ir.WithWidth(this->rank_width, rtl);
Rect icon = ir.Indent(this->rank_width + spacer, rtl).WithWidth(this->icon_size.width, rtl);
Rect text = ir.Indent(this->rank_width + spacer + this->icon_size.width, rtl).WithWidth(this->text_width, rtl);
Rect score = ir.Indent(this->rank_width + spacer + this->icon_size.width + this->text_width, rtl).WithWidth(this->score_width, rtl);
```

`spacer` should be either `hsep_normal` or `hsep_wide`, depending how you want it to look.

This basically ignores `ir.right` / Width, but you've precalculated your widths so that should be okay.